client/service/src/builder.rs

```diff
@@ -638,6 +638,8 @@ impl<TBl, TRtApi, TGen, TCSExt, TCl, TFchr, TSc, TImpQu, TFprb, TFpp, TNetP, TEx
 			Arc<Backend>,
 			Option<TFchr>,
 			Option<Arc<dyn RemoteBlockchain<TBl>>>,
+			Option<TSc>,
+			Arc<RwLock<Keystore>>,
```

Comment:
Can we perhaps pass the entire `&self` to the `rpc_ext_builder` and have accessor methods to retrieve whatever is needed?

Reply:
> have accessor methods to retrieve whatever is needed?

ah this is actually a really great idea. I was a bit worried too about the length of needed deps.

Reply:
I think it's fine as a follow up PR though, don't want to block this one.